dirty_cat/super_vectorizer.py

```diff
@@ -0,0 +1,341 @@
+        if _has_missing_values(X):
+            if self.handle_missing == '':
+                X = _replace_missing(X)
+            elif self.handle_missing == 'error':
+                raise ValueError('Array contains missing values')
+            else:
+                raise ValueError("Invalid 'handle_missing' value. "
+                                 "Expected any of {'', 'error'}, "
+                                 f"got {self.handle_missing}")
+
+        if self.auto_cast:
+            X = self._auto_cast_array(X)
+
+        return X
+
+    def transform(self, X):
+        X = self._transform(X)
+        # [Black magic] - Calls the overridden method `transform`.
+        return ColumnTransformer.transform(self, X)
+
+    def fit_transform(self, X, y=None):
+        X = self._transform(X)
+
+        # Select columns by dtype
+        numeric_columns = X.select_dtypes(include=['int', 'float']).columns.to_list()
+        string_columns = X.select_dtypes(include=['string', 'object']).columns.to_list()
+        categorical_columns = X.select_dtypes(include='category').columns.to_list()
+        datetime_columns = X.select_dtypes(include='datetime').columns.to_list()
+
+        # Divide string and categorical columns by cardinality
+        low_card_str_columns = [col for col in string_columns if X[col].nunique() < self.cardinality_threshold]
+        high_card_str_columns = [col for col in string_columns if X[col].nunique() >= self.cardinality_threshold]
+        low_card_cat_columns = [col for col in categorical_columns if X[col].nunique() < self.cardinality_threshold]
+        high_card_cat_columns = [col for col in categorical_columns if X[col].nunique() >= self.cardinality_threshold]
+
+        # Construct the transformers
+        transformers_df = pd.DataFrame([
+            ['numeric', self.numerical_transformer, numeric_columns],
+            ['datetime', self.datetime_transformer, datetime_columns],
+            ['low_card_str', self.low_card_str_transformer, low_card_str_columns],
+            ['high_card_str', self.high_card_str_transformer, high_card_str_columns],
+            ['low_card_cat', self.low_card_cat_transformer, low_card_cat_columns],
+            ['high_card_cat', self.high_card_cat_transformer, high_card_cat_columns],
+        ], columns=['name', 'transformer', 'columns'])
+        # Get mask of lines with valid encoders.
+        # True: valid encoders
+        valid_encoders_mask = np.invert(transformers_df['transformer'].isnull())
+        # Get mask of lines with non-empty columns
+        # True: non-empty columns field
+        valid_columns_mask = transformers_df['columns'].apply(lambda x: True if x != [] else False)
+
+        mask = valid_encoders_mask & valid_columns_mask
+        self.transformers = transformers_df[mask].values.tolist()
+
+        if not self.transformers:
+            raise RuntimeError('No transformers could be generated !')
+
+        if self.verbose is not False:
```

Comment:
```suggestion
        if self.verbose:
```
"is" in Python tests the exact identity of objects. You seldom want to use it. "verbose=0" is a valid way of specifying no verbosity, and it would not pass the "is" test.

Reply:
Oh, you're right ! I didn't think about the edge case `verbose=0`